test: misc. CI improvements

ianvernon opened the pull request:
This PR performs some miscellaneous cleanup of the tests themselves, as well as adds wrappers around the results of some of the helper functions. This will reduce the amount of boilerplate code in the tests. Furthermore, it refactors some of the helper functions to remove useless return values and groups declarations of `var`s together.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/helpers: fix ManifestGet to use filepath.Join

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/helpers: remove Kubectl receiver from ManifestGet

The receiver was never actually used in the function body, which means it is not
necessary.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: group var declarations in var( ... )

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: move instantiation of vars to when they are declared

Now that ManifestGet is not dependent upon instantiation of a Kubectl structure,
this can be done. It makes the code more readable and organized.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: move K8s chaos test to use BeforeAll

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: add some assertion helpers

These are functions which are wrappers around helper functions and assertions on
their results which are commonly used in the K8s tests. This will reduce the
amount of boilerplate code in the tests.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: get manifests in var declarations

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: have KafkaPolicies test use assertion helpers

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: add wrapper for expecting all pods to be deleted

Add a wrapper which expects WaitCleanAllTerminatingPods to succeed, as these two operations
were always done in conjunction with one another.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: replace WaitKubeDNS with ExpectKubeDNSReady

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: refactor WaitForPods to return only an error

This function previously returned a boolean and an error. However, the boolean was
always false when the error was non-nil, and true when the error was nil, which
makes it useless. Remove it. Also update WaitKubeDNS to reflect this.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: use ExpectCiliumReady in more helpers

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: remove unused demoPath var

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: move instanation of var to its declaration

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: move initialize function for demo test into BeforeAll

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: group var declarations

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: move Health.go initialization into BeforeAll

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: change WaitForServiceEndpoints to only return an error

This function returned a boolean and an error, but the boolean was true if the error
was nil, and false if the error was non-nil, making it useless. Remove it.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: move instantiation of manifest variables in declarations

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: remove unneeded type declarations for vars

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: move instantiation of vars to declaration

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: move initialize function to BeforeAll

Signed-off by: Ian Vernon <ian@cilium.io>

eloycoto commented on test/helpers/kubectl.go:
```diff
@@ -341,13 +341,13 @@ func (kub *Kubectl) Logs(namespace string, pod string) *CmdRes {
 // returns the global manifest file.
-func (kub *Kubectl) ManifestGet(manifestFilename string) string {
-	fullPath := fmt.Sprintf("%s/%s/%s", manifestsPath, GetCurrentK8SEnv(), manifestFilename)
+func ManifestGet(manifestFilename string) string {
```
I'm agree with remove the kubectl here, but I would move this function to another place, `helpers/utils.go` I think that it's the right place.

eloycoto commented on test/k8sT/Services.go:
```diff
@@ -48,15 +48,13 @@ var _ = Describe("K8sValidatedServicesTest", func() {
+		ExpectKubeDNSReady(kubectl)
 
-		ciliumPodK8s1, err = kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+		_, err := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
```
If you delete the CiliumPodK8s1, the following in line 469 will not work no?

```
			By("Checking that policies were correctly imported into Cilium")
			res := kubectl.ExecPodCmd(helpers.KubeSystemNamespace, ciliumPodK8s1, policyCmd)
			res.ExpectSuccess("Policy %s is not imported", policyCmd)
```

Commit:
test/helpers: move ManifestGet to utils.go

Now that ManifestGet does not have a receiver, move it to the file containing
utility functions.

Signed-off by: Ian Vernon <ian@cilium.io>

eloycoto reviewed (approved):
Awesome, thanks!
